lib/signer.js

```diff
@@ -0,0 +1,38 @@
+"use strict";
+
+
+/**
+ * @typedef {signer}
+ * @function sign
+ */
+
+
+/**
+ * Creates signer.
+ *
+ * @param {AuthorizationBuilder} AuthorizationBuilder
+ * @return {signer}
+ */
+module.exports = (AuthorizationBuilder) => {
+    /**
+     * @param {string} code
+     * @param {string} secret
+     * @param {requestOptions} requestOptions
+     * @param {string|stream~Readable} body
+     *
+     * @return {Promise}
+     */
+    function sign(code, secret, requestOptions, body) {
```

Comment:
Also, just to clarify currently the ContentHasher takes a bunch of content whether it is a stream, a string, or Buffer, then when hash is called on it, it concatenates all of the content into one stream and pipes it. So unless I am missing something. This will be a big rework but we will see.